crates/matrix-sdk/src/widget/matrix.rs

```diff
@@ -116,6 +120,16 @@ impl MatrixDriver {
         delayed_event_parameters: Option<delayed_events::DelayParameters>,
     ) -> Result<SendEventResponse> {
         let type_str = event_type.to_string();
+
+        if let Some(redacts) = from_raw_json_value::<Value, serde_json::Error>(&content)
+            .ok()
+            .and_then(|b| b["redacts"].as_str().map(|s| EventId::parse(s).ok()))
+            .flatten()
```

Comment:
Does this work?

```suggestion
            .and_then(|b| b["redacts"].as_str())
            .and_then(|s| EventId::parse(s).ok())
```

Reply:
I dont think so since b is only referenced and does not like leaving the `|b| {}` context.

Reply:
we can do it with one additional clone (to_owned)?

Reply:
I ended up to at least replace the map+flatten with a and_then.

Reply:
```
from_raw_json_value::<Value, serde_json::Error>(&content)
            .ok()
            .and_then(|b| b["redacts"].as_str().and_then(|s| EventId::parse(s).ok()))
            ```